include/aca_dhcp_server.h

```diff
@@ -123,7 +123,8 @@ struct dhcp_client_id {
   uint8_t code;
   uint8_t len;
   uint8_t type;
-  uint8_t cid[0];
+  // uint8_t cid[0];
+  uint8_t *cid = new uint8_t[0];
```

Comment:
I just noticed this (sorry I didn't catch it previously), I am worry that we are allocating memory for a structure creation without the corresponding delete. 

What is the compiler warning for this? Can we surpress this warning if it doesn't make sense?

Reply:
The warning is like this:

In file included from /root/alcor-control-agent/test/gtest/aca_tests.cpp:28:0:
/root/alcor-control-agent/test/../include/aca_dhcp_server.h:126:16: warning: ISO C++ forbids zero-size array ‘cid’ [-Wpedantic]
   uint8_t cid[0];

-----------------------------------------------------------------------
The warning reason is that ISO C++ forbids zero-size array. I think we do not need to fix the warning (considering it wil bring with the problem of memory release).

Reply:
agreed, let's surpress this warning then, please see the below link on the suggestions:

https://stackoverflow.com/questions/18843534/how-to-suppress-compiler-warning-c4373
https://www.fluentcpp.com/2019/08/30/how-to-disable-a-warning-in-cpp/

Reply:
I can't find the warning code (such 4373) for gcc.
In fact, VC++'s warning code of "zero-size array" is 4200, but it doesn't work for gcc/make.

And I found that gcc has its own format for surpressing warnings.
Such as:
---------------------------------------
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"
uint8_t cid[0];
#pragma GCC diagnostic pop
---------------------------------------

Reply:
I can't find the warning code (such as 4373) for gcc.
In fact, VC++'s warning code of "zero-size array" is 4200, but it doesn't work for gcc/make.

And I found that gcc has its own format for surpressing warnings.
Such as:
---------------------------------------
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"
uint8_t cid[0];
#pragma GCC diagnostic pop
---------------------------------------

Reply:
I can't find the warning code (such 4373) for gcc.
In fact, VC++'s warning code of "zero-size array" is 4200, but it doesn't work for gcc/make.

And I found that gcc has its own format for surpressing warnings.
Such as:
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"
uint8_t cid[0];
#pragma GCC diagnostic pop